data-plane/bindings/adapter/src/adapter.rs

```diff
@@ -0,0 +1,1841 @@
+use slim_auth::shared_secret::SharedSecret;
+use slim_auth::traits::TokenProvider; // For get_token() and get_id()
+use slim_config::component::ComponentBuilder;
+use slim_datapath::api::ProtoSessionType;
+use slim_datapath::messages::Name as SlimName;
+use slim_service::Service;
+use slim_service::app::App;
+use slim_service::errors::ServiceError;
+use slim_session::SessionConfig as SlimSessionConfig;
+use slim_session::session_controller::SessionController;
+use slim_session::{Notification, SessionError as SlimSessionError};
+
+use crate::service_ref::{ServiceRef, get_or_init_global_service};
+
+// Re-export uniffi for proc macros
+use uniffi;
+
+// ============================================================================
+// UniFFI Type Definitions
+// ============================================================================
+
+/// Global Tokio runtime for async operations
+static GLOBAL_RUNTIME: OnceLock<Arc<tokio::runtime::Runtime>> = OnceLock::new();
+
+/// Get or initialize the global Tokio runtime
+///
+/// Configured for FFI workloads with:
+/// - Worker threads: 2x CPU cores (to handle blocking operations better)
+/// - Max blocking threads: 512 (allows high concurrency from FFI calls)
+/// - Named threads for easier debugging
+fn get_runtime() -> &'static Arc<tokio::runtime::Runtime> {
+    GLOBAL_RUNTIME.get_or_init(|| {
+        // Calculate optimal worker thread count
+        // Use 2x CPU cores for workloads with blocking operations from FFI
+        let num_workers = std::env::var("SLIM_TOKIO_WORKERS")
+            .ok()
+            .and_then(|s| s.parse().ok())
+            .unwrap_or_else(|| {
+                let cpus = num_cpus::get();
+                (cpus * 2).max(4) // At least 4 workers, preferably 2x CPUs
+            });
+
+        // Allow configurable max blocking threads (default: 512)
+        let max_blocking = std::env::var("SLIM_MAX_BLOCKING_THREADS")
+            .ok()
+            .and_then(|s| s.parse().ok())
+            .unwrap_or(512);
+
+        Arc::new(
+            tokio::runtime::Builder::new_multi_thread()
+                .worker_threads(num_workers)
+                .max_blocking_threads(max_blocking)
+                .thread_name_fn(|| {
+                    static ATOMIC_ID: AtomicUsize = AtomicUsize::new(0);
+                    let id = ATOMIC_ID.fetch_add(1, Ordering::SeqCst);
+                    format!("slim-rt-{}", id)
+                })
+                .enable_all()
+                .build()
+                .expect("Failed to create Tokio runtime"),
+        )
+    })
+}
+
+/// Initialize the crypto provider
+#[uniffi::export]
+pub fn initialize_crypto() {
+    // Crypto initialization happens automatically in slim_auth
+    // Also initialize the global runtime
+    let _ = get_runtime();
+}
+
+/// Get the version of the SLIM bindings
+#[uniffi::export]
+pub fn get_version() -> String {
+    env!("CARGO_PKG_VERSION").to_string()
```

Comment:
Here we might want to give more information such as the commit hash, the date etc.

Check https://github.com/agntcy/slim/blob/main/data-plane/python/bindings/build.rs